constructs/dns-monitor.mdx

```diff
@@ -0,0 +1,230 @@
+---
+title: 'DnsMonitor Construct'
+description: 'Learn how to configure DNS monitors with the Checkly CLI.'
+sidebarTitle: 'DNS Monitor'
+---
+
+import GeneralMonitorOptionsTable from '/snippets/general-monitor-options-table.mdx';
+
+<Tip>
+Learn more about DNS Monitors in [the DNS monitor overview](/detect/uptime-monitoring/dns-monitors/overview).
+</Tip>
+
+Use DNS Monitors to verify that DNS records resolve correctly and to track lookup performance over time.
+
+<Accordion title="Prerequisites">
+Before creating DNS Monitors, ensure you have:
+
+- An initialized Checkly CLI project
+- A domain or hostname you want to monitor
+- Basic understanding of DNS record types (A, AAAA, CNAME, MX, NS, TXT, SOA)
+
+For additional setup information, see [CLI overview](/cli/overview).
+</Accordion>
+
+<CodeGroup>
+
+```ts Basic Example
+import { Frequency, DnsAssertionBuilder, DnsMonitor } from "checkly/constructs"
+
+new DnsMonitor("dns-monitor", {
+  name: "DNS A Record Monitor",
+  activated: true,
+  maxResponseTime: 1000,
+  degradedResponseTime: 500,
+  frequency: Frequency.EVERY_5M,
+  locations: ["us-east-1", "eu-central-1"],
+  request: {
+    query: "example.com",
+    recordType: "A",
+    assertions: [
+      DnsAssertionBuilder.jsonAnswer("$.Answer[0].data").equals("93.184.216.34")
+    ],
+  },
+})
+```
+
+```ts Advanced Example
+import { Frequency, DnsAssertionBuilder, DnsMonitor } from "checkly/constructs"
+
+new DnsMonitor("advanced-dns-monitor", {
+  name: "Advanced DNS Monitor",
+  activated: true,
+  frequency: Frequency.EVERY_5M,
+  locations: ["us-east-1", "eu-central-1"],
+  maxResponseTime: 800,
+  degradedResponseTime: 400,
+  tags: ["dns", "production", "critical"],
+  request: {
+    query: "api.example.com",
+    recordType: "A",
+    nameServer: "9.9.9.9",
+    port: 53,
+    protocol: "UDP",
+    assertions: [
+      DnsAssertionBuilder.jsonAnswer("$.Answer[0].TTL").between(300, 86400),
```

Comment:
We do not have a `.between(a, b)` operator. If you wanted to validate a range, you'd have to add both a `.lessThan` and a `.greaterThan` to emulate it.